app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js

```diff
@@ -133,11 +133,12 @@ const ErrorMessage = styled.span`
   font-size: 12px;
 `
 
-function* fileUploadGenerator(files, uploadFile) {
+function* fileUploadGenerator(files, uploadFile, removeSingleFile) {
```

Comment:
Could you explain why the client needs to know about "removeSingleFile" ? - thanks

Reply:
Client needs to know whether one file is removed ( replaced for now ) or all of them, as the only option to remove the files for now is to remove them all. But when we upload a repeated file we want to remove that single file and then upload the new version of that, we also need to show the user that this file is changed by adding the loading state as the normal behaviour. Also there are errors that are only defined in the client side, such as `max size exceeded` we want to show that when they try to upload a large repeated file. We could have also name these `removeFile` and `removeFiles`... if we will keep both in the future.

Reply:
We'll eventually be replacing the 'remove all' functionality with the option to remove one file at a time.